Examples/Detectors/Geant4Detector/src/MockupSectorBuilder.cpp

```diff
@@ -108,31 +106,34 @@ ActsExamples::MockupSectorBuilder::BuildChamber(
         detectorElement->transform(context));
 
     straw_surfaces.push_back(g4conv_surf);
-    positionY.push_back(g4conv_surf->center(context).y());
-    positionZ.push_back(g4conv_surf->center(context).z());
-    positionX.push_back(g4conv_surf->center(context).x());
+
+    min_max[0].first = std::min(min_max[0].first, (float) g4conv_surf->center(context).x());
+    min_max[0].second = std::max(min_max[0].second, (float) g4conv_surf->center(context).x());
+
+    min_max[1].first = std::min(min_max[1].first, (float) g4conv_surf->center(context).y());
+    min_max[1].second = std::max(min_max[1].second, (float) g4conv_surf->center(context).y());
+
+    min_max[2].first = std::min(min_max[2].first, (float) g4conv_surf->center(context).z());
+    min_max[2].second = std::max(min_max[2].second, (float) g4conv_surf->center(context).z());
+
   }
 
   // Create the bounds of the detector volumes
   float radius = straw_surfaces.front()->bounds().values()[0];
-  auto minY = *min_element(positionY.begin(), positionY.end());
-  auto maxY = *max_element(positionY.begin(), positionY.end());
-
-  auto minZ = *min_element(positionZ.begin(), positionZ.end());
-  auto maxZ = *max_element(positionZ.begin(), positionZ.end());
+  
+  Acts::Vector3 minValues = {min_max[0].first, min_max[1].first, min_max[2].first};
+  Acts::Vector3 maxValues = {min_max[0].second, min_max[1].second, min_max[2].second};
 
-  auto minX = *min_element(positionX.begin(), positionX.end());
-  auto maxX = *max_element(positionX.begin(), positionX.end());
 
-  Acts::ActsScalar hx = straw_surfaces.front()->bounds().values()[1] + 10;
-  Acts::ActsScalar hy = 0.5 * (std::abs(minY - maxY) + 2 * radius + 10.);
-  Acts::ActsScalar hz = 0.5 * (std::abs(minZ - maxZ) + 2 * radius + 10.);
+  Acts::ActsScalar hx = straw_surfaces.front()->bounds().values()[1] + m_cfg.toleranceOverlap;
+  Acts::ActsScalar hy = 0.5 * (std::abs(minValues.y() - maxValues.y()) + 2 * radius + m_cfg.toleranceOverlap);
+  Acts::ActsScalar hz = 0.5 * (std::abs(minValues.z() - maxValues.z()) + 2 * radius + m_cfg.toleranceOverlap);
```

Comment:
Why do you have a factor 2 here?
Another comment is on the tolerance parameter. Do you think it is better to have one for the cuboid volume and one for the cylinder?

Reply:
I have the factor 2 because we have one radius from the innermost surface plus one from the outermost,, since minY and maxY give me the centers of the surfaces. 
I am not sure about the the tolerance. I  think it is ok to keep it general and use it for both cases? But you can tell me what you think is better

Reply:
```suggestion
  Acts::ActsScalar hy = 0.5 * ((maxValues.y()+radius)-(minValues.y()-radius) + m_cfg.toleranceOverlap);
  Acts::ActsScalar hz = 0.5 * ((maxValues.z()+radius)-(minValues.z()-radius) + m_cfg.toleranceOverlap);
```